content/en/docs/tasks/administer-cluster/nodelocaldns.md

```diff
@@ -49,7 +49,7 @@ This is the path followed by DNS Queries after NodeLocal DNSCache is enabled:
 {{< figure src="/images/docs/nodelocaldns.svg" alt="NodeLocal DNSCache flow" title="Nodelocal DNSCache flow" caption="This image shows how NodeLocal DNSCache handles DNS queries." >}}
 
 ## Configuration
-{{< note >}} The local listen IP address for NodeLocal DNSCache can be any IP in the 169.254.20.0/16 space or any other IP address that can be guaranteed to not collide with any existing IP. This document uses 169.254.20.10 as an example.
+{{< note >}} The local listen IP address for NodeLocal DNSCache can be any link-local address, 169.254.0.0/16 in the IPv4 or fe80::/10 in the IPv6 space, or any other IP address that can be guaranteed to not collide with any existing IP. This document uses 169.254.20.10 as an IPv4 example, for IPv6 you could use fe80::169:254:20:2f17, per example.
```

Comment:
This is valid but, I think, misleading. Bear in mind that IPv6 link-local addresses require an interface identifier and that (depending on the network plugin) each Pod may have a unique link.

I'm not sure how much this is a documentation issue or a design challenge. Essentially, using link-local addresses doesn't fit any network plugin implementation that uses a unique link per Pod, even though that network architecture is completely valid. With IPv4 you can hide this detail behind routeing rules whereas with IPv6 it's more trouble.

Maybe this discussion belongs elsewhere? I'm happy to continue explaining my thinking outside the documentation repo.

Reply:
If the collision and interface ID issues are sorted, how about suggesting `fe80::53`?

Reply:
you are absolutely right, I had to check the RFC https://tools.ietf.org/html/rfc4291 because I honestly didn't remember this limitation of the link-local, I've remember to modifed the routers to not use the mac generated link-local but never realized that there has to have always an inteface identigies.
Thanks for pointing it out and happy to collaborate with you in defining a standarized model for these kind of deployments

Reply:
> If the collision and interface ID issues are sorted, how about suggesting `fe80::53`?

I would like to explore the zones/scope option https://tools.ietf.org/html/rfc4007,  what do you think?

Reply:
Should this be an addition to the nodelocaldns KEP?

Reply:
yeah, let's do it right.

Reply:
I have to admit I wasn' t fully understanding how nodelocadns cache works
It works right now with IPv6, the only thing that doesn' t work are the iptables rules, but there aren' t any architecural changes needed

Reply:
There are no architectural changes needed. But don't we need some guideline on what IP address to use, or is that left for the user to decide? They need to pick an IP that will work on all nodes, since it is a common daemonset config.

Reply:
If you are planning to merge what is in this PR, that sounds fine to me. There is some guideline on what ip to use and is left to the user to make the choice.